src/parser/TerraformListener.js

```diff
@@ -226,40 +226,79 @@ class TerraformListener extends antlr4.tree.ParseTreeListener {
         this.currentField.name,
       );
     }
-
     this.currentField = null;
   }
 
   // Enter a parse tree produced by terraformParser#complexField.
-  enterComplexField(ctx) {
+  enterComplexField() {
+  }
+
+  // Exit a parse tree produced by terraformParser#complexField.
+  exitComplexField() {
+  }
+
+  // Enter a parse tree produced by terraformParser#dynamicBlock.
+  enterDynamicBlock(ctx) {
     const name = getText(ctx.IDENTIFIER());
     let definition;
 
-    if (this.currentComplexeField !== null) {
-      this.fieldsTree.push(this.currentComplexeField);
-      definition = this.getAttributeDefinition(this.currentComplexeField, name);
+    if (this.currentComplexField !== null) {
+      this.fieldsTree.push(this.currentComplexField);
+      definition = this.getAttributeDefinition(this.currentComplexField, name);
     } else {
       definition = this.getAttributeDefinition(this.currentComponent, name);
     }
+    this.currentComplexField = new TerraformComponentAttribute({
+      name,
+      value: [],
+      type: 'Object',
+      definition,
+      isDynamic: true,
+    });
+  }
+
+  // Exit a parse tree produced by terraformParser#dynamicBlock.
+  exitDynamicBlock() {
+    if (this.fieldsTree.length > 0) {
+      const field = this.fieldsTree.pop();
+
+      field.value.push(this.currentComplexField);
+      this.currentComplexField = field;
```

Comment:
complexeField no more exist in grammar, so can you rename this variable in `currentObjectField`

Reply:
Done